File: Source/QText/MenuStripExOnMainForm.vb
Friend Class MenuStripExOnMainForm : Inherits System.Windows.Forms.MenuStrip

    Private Sub MenuStripEx_KeyUp(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles Me.KeyUp
        Debug.WriteLine("MenuStripEx.KeyUp: " + e.KeyData.ToString())
        Dim mainForm As MainForm = TryCast(Me.Parent, MainForm)
        If (mainForm Is Nothing) Then Return
        Dim tabFiles As QTextAux.TabControlDnD = mainForm.tabFiles

        Select Case e.KeyData
            Case Keys.Menu
                If (Me.Visible) Then
                    If (Not Settings.ShowMenu) Then
                        Me.Visible = False
                    End If
                    If (tabFiles.SelectedTab IsNot Nothing) Then
                        Dim txt As TextBoxBase = DirectCast(tabFiles.SelectedTab, QTextAux.TabFile).TextBox
                        txt.Focus()
                    End If
                End If

        End Select
    End Sub

    Private Sub MenuStripEx_MenuDeactivate(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.MenuDeactivate
        Debug.WriteLine("MenuStripEx.Deactivate")
        Dim mainForm As MainForm = TryCast(Me.Parent, MainForm)
        If (mainForm Is Nothing) Then Return
        Dim tabFiles As QTextAux.TabControlDnD = mainForm.tabFiles

        Me.Visible = Settings.ShowMenu

        If (tabFiles.SelectedTab IsNot Nothing) Then
            Dim txt As TextBoxBase = DirectCast(tabFiles.SelectedTab, QTextAux.TabFile).TextBox
            txt.Focus()
        End If
    End Sub

End Class

File: Source/QText/MenuStripExOnMainForm.cs
using System;
using System.Windows.Forms;

namespace QText {
    internal class MenuStripExOnMainForm : MenuStrip {

        public MenuStripExOnMainForm() {
            this.KeyUp += new KeyEventHandler(MenuStripExOnMainForm_KeyUp);
            this.MenuDeactivate += new EventHandler(MenuStripExOnMainForm_MenuDeactivate);
        }


        private void MenuStripExOnMainForm_KeyUp(object sender, System.Windows.Forms.KeyEventArgs e) {
            MainForm mainForm = this.Parent as MainForm;
            if (mainForm == null) { return; }
            TabFiles tabFiles = mainForm.tabFiles;

            switch (e.KeyData) {
                case Keys.Menu:
                    if (this.Visible) {
                        if (tabFiles.SelectedTab != null) {
                            TextBoxBase txt = tabFiles.SelectedTab.TextBox;
                            txt.Focus();
                        }
                    }

                    break;
            }
        }

        private void MenuStripExOnMainForm_MenuDeactivate(object sender, System.EventArgs e) {
            MainForm mainForm = this.Parent as MainForm;
            if (mainForm == null) { return; }
            TabFiles tabFiles = mainForm.tabFiles;

            if (tabFiles.SelectedTab != null) {
                TextBoxBase txt = tabFiles.SelectedTab.TextBox;
                txt.Focus();
            }
        }

    }
}
